tabular/src/autogluon/tabular/models/automm/automm_model.py

```diff
@@ -0,0 +1,216 @@
+        path
+            The directory to store the modeling outputs.
+        name
+            Name of subdirectory inside path where model will be saved.
+        problem_type
+            Type of problem that this model will handle.
+            Valid options: ['binary', 'multiclass', 'regression'].
+        eval_metric
+            The evaluation metric.
+        num_classes
+            The number of classes.
+        stopping_metric
+            The stopping metric.
+        model
+            The internal model object.
+        hyperparameters
+            The hyperparameters of the model
+        features
+            Names of the features.
+        feature_metadata
+            The feature metadata.
+        """
+        super().__init__(**kwargs)
+        self._label_column_name = None
+        self._load_model = None  # Whether to load inner model when loading.
+
+    def _get_default_auxiliary_params(self) -> dict:
+        default_auxiliary_params = super()._get_default_auxiliary_params()
+        extra_auxiliary_params = dict(
+            valid_raw_types=[R_INT, R_FLOAT, R_CATEGORY, R_OBJECT],
+            ignored_type_group_special=[S_TEXT_NGRAM, S_TEXT_AS_CATEGORY, S_TEXT_SPECIAL],
+        )
+        default_auxiliary_params.update(extra_auxiliary_params)
+        return default_auxiliary_params
+
+    @classmethod
+    def _get_default_ag_args(cls) -> dict:
+        default_ag_args = super()._get_default_ag_args()
+        extra_ag_args = {'valid_stacker': False}
+        default_ag_args.update(extra_ag_args)
+        return default_ag_args
+
+    def _set_default_params(self):
+        super()._set_default_params()
+        try_import_autogluon_text()
+
+    def _fit(self,
+             X: pd.DataFrame,
+             y: pd.Series,
+             X_val: Optional[pd.DataFrame] = None,
+             y_val: Optional[pd.Series] = None,
+             time_limit: Optional[int] = None,
+             sample_weight=None,
+             **kwargs):
+        """The internal fit function
+
+        Parameters
+        ----------
+        X
+            Features of the training dataset
+        y
+            Labels of the training dataset
+        X_val
+            Features of the validation dataset
+        y_val
+            Labels of the validation dataset
+        time_limit
+            The time limits for the fit function
+        sample_weight
+            The weights of the samples
+        kwargs
+            Other keyword arguments
+
+        """
+        try_import_autogluon_text()
+        from autogluon.text.automm import AutoMMPredictor
+
+        # Decide name of the label column
+        if 'label' in X.columns:
+            label_col_id = 0
+            while True:
+                self._label_column_name = 'label{}'.format(label_col_id)
+                if self._label_column_name not in X.columns:
+                    break
+                label_col_id += 1
+        else:
+            self._label_column_name = 'label'
```

Comment:
Maybe worth moving to a dedicated method for better readability

Also: This should be after `self.preprocess` calls, to ensure that the column name wasn't somehow added during preprocessing.